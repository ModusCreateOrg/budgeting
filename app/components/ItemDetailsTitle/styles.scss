  h3, p {
    text-align: center
  }

  .negative {
    h3 {
      font-weight: bold;
    }
    p {
      color: red;
    }
  }

  .positive {
    h3 {
      font-weight: bold;
    }
    p {
      color: blue;
    }
  }