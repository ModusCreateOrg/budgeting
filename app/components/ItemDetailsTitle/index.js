// @flow
import React from 'react';
import styles from './styles.scss';

type ItemDetailtsTitleProps = {
  title: string,
  subtitle: string,
  type: Boolean,
};

const ItemDetailtsTitle = ({ title, subtitle, type }): ItemDetailtsTitleProps => (
  <div className={type ? styles.negative : styles.positive}>
    <h3>{title}</h3>
    <p>{subtitle}</p>
  </div>
);

export default ItemDetailtsTitle;
