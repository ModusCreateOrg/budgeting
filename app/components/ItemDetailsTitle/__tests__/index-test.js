import React from 'react';
import renderer from 'react-test-renderer';
import ItemDetailsTitle from '../';

const ItemDetailsTitleProps = {
  description: 'I am a transaction.',
  subtitle: 'is the 22% of the outflows',
  isNegative: true,
};

it('renders correctly', () => {
  const tree = renderer
    .create(
      <ItemDetailsTitle
        title={ItemDetailsTitleProps.description}
        subtitle={ItemDetailsTitleProps.subtitle}
        type={ItemDetailsTitleProps.isNegative}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
