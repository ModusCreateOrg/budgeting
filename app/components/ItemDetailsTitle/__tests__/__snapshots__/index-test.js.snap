// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="negative"
>
  <h3>
    I am a transaction.
  </h3>
  <p>
    is the 22% of the outflows
  </p>
</div>
`;
